stellargraph/layer/rgcn.py

```diff
@@ -0,0 +1,496 @@
+
+        super().__init__(**kwargs)
+
+        self.units = units
+        self.num_relationships = num_relationships
+        self.num_bases = num_bases
+        self.activation = activations.get(activation)
+        self.use_bias = use_bias
+        self.kernel_initializer = initializers.get(kernel_initializer)
+        self.bias_initializer = initializers.get(bias_initializer)
+        self.kernel_regularizer = regularizers.get(kernel_regularizer)
+        self.bias_regularizer = regularizers.get(bias_regularizer)
+        self.activity_regularizer = regularizers.get(activity_regularizer)
+        self.kernel_constraint = constraints.get(kernel_constraint)
+        self.bias_constraint = constraints.get(bias_constraint)
+
+        self.final_layer = final_layer
+
+    def get_config(self):
+        """
+        Gets class configuration for Keras serialization.
+        Used by keras model serialization.
+
+        Returns:
+            A dictionary that contains the config of the layer
+        """
+
+        config = {
+            "units": self.units,
+            "use_bias": self.use_bias,
+            "final_layer": self.final_layer,
+            "activation": activations.serialize(self.activation),
+            "kernel_initializer": initializers.serialize(self.kernel_initializer),
+            "bias_initializer": initializers.serialize(self.bias_initializer),
+            "kernel_regularizer": regularizers.serialize(self.kernel_regularizer),
+            "bias_regularizer": regularizers.serialize(self.bias_regularizer),
+            "activity_regularizer": regularizers.serialize(self.activity_regularizer),
+            "kernel_constraint": constraints.serialize(self.kernel_constraint),
+            "bias_constraint": constraints.serialize(self.bias_constraint),
+            "num_relationships": self.num_relationships,
+            "num_bases": self.num_bases,
+        }
+
+        base_config = super().get_config()
+        return {**base_config, **config}
+
+    def compute_output_shape(self, input_shapes):
+        """
+        Computes the output shape of the layer.
+
+        Args:
+            input_shapes (tuple of ints)
+                Shape tuples can include None for free dimensions, instead of an integer.
+
+        Returns:
+            An input shape tuple.
+        """
+        feature_shape, out_shape, A_shape = input_shapes
+
+        batch_dim = feature_shape[0]
+        if self.final_layer:
+            out_dim = out_shape[1]
+        else:
+            out_dim = feature_shape[1]
+
+        return batch_dim, out_dim, self.units
+
+    def build(self, input_shapes):
+        """
+        Builds the layer
+
+        Args:
+            input_shapes (list of int): shapes of the layer's inputs
+            (node features, node_indices, and adjacency matrices)
+
+        """
+        feat_shape = input_shapes[0]
+        input_dim = int(feat_shape[-1])
+
+        if self.num_bases != 0:
```

Comment:
I guess negative values are good here!

Maybe change it to self.num_bases > 0 so that at least negative values are treated as zero. A check for a valid value and a unit test to go with it would be helpful.

Reply:
Nice pickup! I'll change it to `self.num_bases > 0` and add a unit test around this